src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -422,6 +422,7 @@ public interface IStartConfiguration
             IAuthProvider AuthProvider { get; set; }
             IEnumerable<IExtension> Extensions { get; set; }
             TaskProcessMode ProcessMode { get; set; }
+            bool HeadlessMode { get; set; }
```

Comment:
Unfortunately, it doesn't look like that would help.  The CheckForProductUpdate static function does too much of the work by itself.